src/components/DataTable/DataTable.tsx

```diff
@@ -230,99 +162,48 @@ class DataTable extends React.PureComponent<CombinedProps, DataTableState> {
                 {totalsMarkup}
               </thead>
               <tbody>{bodyMarkup}</tbody>
-              {footerMarkup}
             </table>
           </div>
+          {footerMarkup}
         </div>
       </div>
     );
   }
 
-  private tallestCellHeights = () => {
-    const {footerContent, truncate} = this.props;
-    const {
-      table: {current: table},
-    } = this;
-    let {heights} = this.state;
-    if (table) {
-      const rows = Array.from(table.getElementsByTagName('tr'));
-
-      if (!truncate) {
-        return (heights = rows.map((row) => {
-          const fixedCell = (row.childNodes as NodeListOf<HTMLElement>)[0];
-          return Math.max(row.clientHeight, fixedCell.clientHeight);
-        }));
-      }
-
-      if (footerContent) {
-        const footerCellHeight = (rows[rows.length - 1]
-          .childNodes as NodeListOf<HTMLElement>)[0].clientHeight;
-        heights = [footerCellHeight];
-      }
-    }
-
-    return heights;
-  };
-
-  private resetScrollPosition = () => {
-    const {
-      scrollContainer: {current: scrollContainer},
-    } = this;
-    if (scrollContainer) {
-      const {
-        preservedScrollPosition: {left, top},
-      } = this.state;
-      if (left) {
-        scrollContainer.scrollLeft = left;
-      }
-      if (top) {
-        window.scrollTo(0, top);
-      }
-    }
-  };
-
-  private setHeightsAndScrollPosition = () => {
-    this.setState(
-      {heights: this.tallestCellHeights()},
-      this.resetScrollPosition,
-    );
-  };
-
-  private calculateColumnVisibilityData = (collapsed: boolean) => {
+  private calculateColumnVisibilityData = (condensed: boolean) => {
     const {
       table: {current: table},
       scrollContainer: {current: scrollContainer},
       dataTable: {current: dataTable},
     } = this;
-    if (collapsed && table && scrollContainer && dataTable) {
+
+    if (condensed && table && scrollContainer && dataTable) {
       const headerCells = table.querySelectorAll(
         headerCell.selector,
       ) as NodeListOf<HTMLElement>;
-      const collapsedHeaderCells = Array.from(headerCells).slice(1);
-      const fixedColumnWidth = headerCells[0].offsetWidth;
-      const firstVisibleColumnIndex = collapsedHeaderCells.length - 1;
-      const tableLeftVisibleEdge =
-        scrollContainer.scrollLeft + fixedColumnWidth;
+
+      const firstVisibleColumnIndex = headerCells.length - 1;
+      const tableLeftVisibleEdge = scrollContainer.scrollLeft;
+
       const tableRightVisibleEdge =
         scrollContainer.scrollLeft + dataTable.offsetWidth;
+
       const tableData = {
-        fixedColumnWidth,
         firstVisibleColumnIndex,
         tableLeftVisibleEdge,
         tableRightVisibleEdge,
       };
 
-      const columnVisibilityData = collapsedHeaderCells.map(
+      const columnVisibilityData = Array.from(headerCells).map(
```

Comment:
Array.from is used a few times here. Judging by this https://www.measurethat.net/Benchmarks/ShowResult/59877looks like spreading is more performant.

Reply:
We have to change `headerCells` from a NodeList to an Array, it's not spreadable unfortunately.

Reply:
You know what scratch that. I just tried it and TS doesn't yell about it anymore!